mmv1/third_party/terraform/utils/node_config.go.erb

```diff
@@ -265,9 +267,17 @@ func schemaNodeConfig() *schema.Schema {
 						Schema: map[string]*schema.Schema{
 							"node_metadata": {
 								Type:         schema.TypeString,
-								Required:     true,
+								Optional:     true,
+								Computed:     true,
+								Deprecated:   "Deprecated in favor of mode.",
 								ValidateFunc: validation.StringInSlice([]string{"UNSPECIFIED", "SECURE", "EXPOSE", "GKE_METADATA_SERVER"}, false),
-								Description: `NodeMetadata is the configuration for how to expose metadata to the workloads running on the node.`,
+								Description:  `NodeMetadata is the configuration for how to expose metadata to the workloads running on the node.`,
+							},
+							"mode": {
+								Type:         schema.TypeString,
+								Optional:     true,
```

Comment:
Will the API return this field if `node_metadata` is set? We probably need `Computed: true` if so.

Reply:
I just tested applying `mode` first, and then replacing with `node_metadata`. It did not return `mode`.